pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -752,20 +751,11 @@ func removeTerminatedContainers(containers []*runtimeapi.Container) []*runtimeap
 
 	result := make([]*runtimeapi.Container, 0)
 	for _, refs := range containerMap {
-		if len(refs) == 1 {
-			result = append(result, refs[0])
-			continue
-		}
-		found := false
 		for i := 0; i < len(refs); i++ {
 			if refs[i].State == runtimeapi.ContainerState_CONTAINER_RUNNING {
```

Comment:
Nitpick: “terminated” and “not running” aren’t the same thing: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-states